### PR TITLE
fix: added SKU to the products array

### PR DESCRIPTION
**Pull request by dbelkad**
### PR CHECKLIST

- [x] All translations are concatenated with an EOT character between the translation context and key (ie. `backend/config[EOT]general_settings_header`)
- [x] The version information documented in the readme.txt has been manually updated in line with semantic versioning
- [x] The readme file's Changelog has been updated with your proposed PR
- [x] The version information in `class-wc-gateway-finance.php` (included within the initial docblock and the start of the constructor) has been updated

### CODE REVIEW

File: class-wc-gateway-finance.php [hunk 1]
```diff
@@ -1372,78 +1371,86 @@ function process_payment($order_id)
...
+                  $price = $item['line_subtotal'] / $quantity * 100;
+                  $order_total += $item['line_subtotal'];
+                  $products[] = array(
+                    'name' => $name,
+                    'quantity' => (int) $quantity,
+                    'price' => round($price),
+                    'sku' => $item['data']->get_sku() ?? $item['data']->get_id()
+                  );
                 }
+                print_r($products);exit;
```
Comment (andivido) [inline class-wc-gateway-finance.php:1391]:
```suggestion
                
```

File: class-wc-gateway-finance.php [hunk 2]
```diff
@@ -787,7 +787,7 @@ public function product_widget($product)
                 $price = $this->get_price_including_tax($product, '');
                 $plans = $this->get_product_plans($product);
                 $environment = $this->get_finance_env();
-                if ($this->is_available($product) && $price > ($this->widget_threshold * 100)) {
+                if ($this->is_available($product) && $price > (((int)$this->widget_threshold ?? 0) * 100)) {
```
Comment (andivido) [inline class-wc-gateway-finance.php:790]: Just need to make sure we don't overwrite our other PR

**Review by andivido — CHANGES_REQUESTED** [review]
Looks good. Just need to remove the die dump. Try to put formatting changes in a separate commit, so it's easier to see what material changes were carried out

File: class-wc-gateway-finance.php (hunk 2 above)
Comment (dbelkad) [reply class-wc-gateway-finance.php:790]: It shouldn't. And if there is a conflict, I'll fix it before merging the PR.

**Review by andivido — APPROVED** [review]

File: readme.txt [hunk 3]
```diff
@@ -1,194 +1,194 @@
...
+Contributors:      Divido Financial Services Ltd
+Plugin Name:       Finance Gateway for WooCommerce
+Plugin URI:        integrations.divido.com/woocommerce
+Tags:              woothemes,woocommerce,payment gateway,payment,module,ecommerce,online payments,
+Author URI:        integrations.divido.com
+Author:            Divido Financial Services Ltd
+Requires at least: 3.0.2
+Tested up to:      5.7.2
+Stable tag:        2.3.3
+Version:           2.3.3
```
Comment (andivido) [inline readme.txt:11]: These need to be bumped to 2.3.4 now

File: readme.txt [hunk 4]
```diff
@@ -1,194 +1,194 @@
...
+Enable/Disable Automatic Fulfillment: Allows you to select if an "Activation" call should be made automatically to the lender once the order goes to "Completed"
+Enable/Disable Automatic Refunds: Allows you to select if a "Refund" call should be made automatically to the lender once the order goes to "Refunded"
+Enable/Disable Automatic Cancellation: Allows you to select if an "Cancellation" call should be made automatically to the lender once the order goes to "Cancelled"
+
+
+
+ == Changelog ==
+Version 2.3.4
+Fix: Adding SKU to the products array.
+
```
Comment (andivido) [inline readme.txt:49]: you seemed to have lost the changelog for 2.3.3 somewhere

**Review by andivido — CHANGES_REQUESTED** [review]

File: readme.txt [hunk 5]
```diff
@@ -1,194 +1,194 @@
-=== Plugin Name ===
```
Comment (joelbitar) [inline readme.txt:1]: Why is the entire file marked as updated, seems weird to me...

**Review by joelbitar — CHANGES_REQUESTED** [review]
We should try to avoid to change formatting when we don't need to. Makes it a lot harder to follow along and what changes actually matter, accidentally introducing new bugs.
Could we try to just change the lines that needs changing?

File: readme.txt (hunk 4 above)
Comment (dbelkad) [reply readme.txt:49]: I worked on the fix for the cart threshold where you can see the changes in log to 2.3.3. This was another task, hence the another bump as the PR hasn't been merged yet.

File: readme.txt (hunk 5 above)
Comment (dbelkad) [reply readme.txt:1]: I'm sorry about the re-formatting. It's a bad habit of mine. I'll revert the changes.

**Pull request closed without merging**